src/idpi/operators/regrid.py

```diff
@@ -0,0 +1,184 @@
+    )
+
+
+def _relimit(longitude: float) -> float:
+    if longitude > 180:
+        return longitude - 360
+    return longitude
+
+
+@dc.dataclass
+class RegularGrid:
+    """Class defining a regular grid.
+
+    Attributes
+    ----------
+    crs : CRS
+        Coordinate reference system.
+    nx : int
+        Number of grid points in the x direction.
+    ny : int
+        Number of grid points in the y direction.
+    xmin : float
+        Coordinate of the first grid point in the x direction.
+    xmax : float
+        Coordinate of the last grid point in the x direction.
+    ymin : float
+        Coordinate of the first grid point in the y direction.
+    ymax : float
+        Coordinate of the last grid point in the y direction.
+
+    """
+
+    crs: CRS
+    nx: int
+    ny: int
+    xmin: float
+    xmax: float
+    ymin: float
+    ymax: float
+
+    @classmethod
+    def from_field(cls, field: xr.DataArray):
+        """Extract grid parameters from grib metadata.
+
+        Parameters
+        ----------
+        field : xarray.DataArray
+            field containing the relevant metadata.
+
+        """
+        geo = field.geography
+        obj = cls(
+            crs=_get_crs(geo),
+            nx=geo["Ni"],
+            ny=geo["Nj"],
+            xmin=_relimit(geo["longitudeOfFirstGridPointInDegrees"]),
+            xmax=_relimit(geo["longitudeOfLastGridPointInDegrees"]),
+            ymin=geo["latitudeOfFirstGridPointInDegrees"],
+            ymax=geo["latitudeOfLastGridPointInDegrees"],
+        )
+        if abs(obj.dx - geo["iDirectionIncrementInDegrees"]) > 1e-5:
+            raise ValueError("Inconsistent grid parameters")
+        if abs(obj.dy - geo["jDirectionIncrementInDegrees"]) > 1e-5:
+            raise ValueError("Inconsistent grid parameters")
+        return obj
+
+    @classmethod
+    def parse_regrid_operator(cls, op: str):
+        """Parse fieldextra out_regrid_target string.
+
+        Parameters
+        ----------
+        op : str
+            fieldextra out_regrid_target definition
+            i.e. crs,xmin,ymin,xmay,ymax,dx,dy.
+
+        """
+        crs_str, *grid_params = op.split(",")
+        crs = CRS.from_string(CRS_ALIASES[crs_str])
+        xmin, ymin, xmax, ymax, dx, dy = map(float, grid_params)
+        nx = (xmax - xmin) / dx + 1
+        ny = (ymax - ymin) / dy + 1
+        if nx != int(nx) or ny != int(ny):
+            raise ValueError("Inconsistent regrid parameters")
+        return cls(crs, int(nx), int(ny), xmin, xmax, ymin, ymax)
+
+    @property
+    def dx(self) -> float:
+        return (self.xmax - self.xmin) / (self.nx - 1)
+
+    @property
+    def dy(self) -> float:
+        return (self.ymax - self.ymin) / (self.ny - 1)
+
+    @property
+    def transform(self) -> transform.Affine:
+        return transform.from_origin(
+            west=self.xmin - self.dx / 2,
+            north=self.ymax + self.dy / 2,
```

Comment:
why half a delta?

Reply:
I think it's okay to leave that little bit of uncertainty in the code. It seems to work and we can always improve it when we learn more about it. LGTM